File: CrossFiles/Python/class2.py
import sqlite3

class Class2:
    def __init__(self, input_data):
        self.input = input_data
        self.db = None

    def connect_db(self, db_name):
        try:
            self.db = sqlite3.connect(db_name)
            return True
        except sqlite3.Error as e:
            print(f"Error connecting to database: {e}")
            return False

    def process(self):
        print(f"Class2 processing: {self.input}")
        if not self.db:
            print("Database not connected")
            return

        # Potentially unsafe operation
        query = f"SELECT * FROM users WHERE username = '{self.input}'"
        print(f"Executing query: {query}")

        try:
            cursor = self.db.cursor()
            cursor.execute(query)
            results = cursor.fetchall()
            print(f"Query results: {results}")
        except sqlite3.Error as e:
            print(f"Error executing query: {e}")

    def __del__(self):
        if self.db:
            self.db.close()

File: CrossFiles/JavaScript/class2.js
const sqlite3 = require('sqlite3').verbose();

class Class2 {
    constructor(input) {
        this.input = input;
        this.db = null;
    }

    connectDb(dbName) {
        return new Promise((resolve, reject) => {
            this.db = new sqlite3.Database(dbName, (err) => {
                if (err) {
                    console.error('Could not connect to database', err);
                    reject(err);
                } else {
                    resolve();
                }
            });
        });
    }

    process() {
        console.log(`Class2 processing: ${this.input}`);
        if (!this.db) {
            console.log("Database not connected");
            return;
        }

        // Potentially unsafe operation
        const query = `SELECT * FROM users WHERE username = '${this.input}'`;
        console.log(`Executing query: ${query}`);

        this.db.all(query, [], (err, rows) => {
            if (err) {
                console.error("Error executing query:", err);
            } else {
                console.log("Query results:", rows);
            }
        });
    }

    closeDb() {
        if (this.db) {
            this.db.close();
        }
    }
}

module.exports = Class2;
